modules/service/src/main/scala/lucuma/odb/service/ExposureTimeModeService.scala

```diff
@@ -159,65 +205,115 @@ object ExposureTimeModeService:
-
-        def success: Map[Observation.Id, (ExposureTimeMode, ExposureTimeMode)] =
-          sci.fproductLeft: etm =>
-            explicitAcquisition.getOrElse(ExposureTimeMode.forAcquisition(etm.at))
-
-        def error: Problem =
-          val msg  = s"${observingModeName} requires a science exposure time mode."
-          OdbError.InvalidArgument(msg.some).asProblem
-
-        Result.fromOption(Option.when(missing.isEmpty)(success), error)
-
-      override def insertForSingleScienceEtm(
+      override def resolve[A](
         observingModeName:   String,
         explicitAcquisition: Option[ExposureTimeMode],
-        explicitScience:     Option[ExposureTimeMode],
+        explicitScience:     NonEmptyList[(A, Option[ExposureTimeMode])],
         newRequirement:      Option[ExposureTimeMode],
         which:               List[Observation.Id]
-      )(using Transaction[F]): F[Result[Unit]] =
-        def insert(etms: List[(Observation.Id, ExposureTimeMode)], role: ExposureTimeModeRole): F[Unit] =
-          etms.groupMap(_._2)(_._1).toList.traverse_ { (etm, oids) =>
-            services.exposureTimeModeService.insertMany(NonEmptyList.fromListUnsafe(oids), role, etm)
-          }
-
-        NonEmptyList
-          .fromList(which)
-          .fold(Result.unit.pure[F]): nel =>
-            (for
-              c <- ResultT.liftF(selectRequirement(nel))
-              m <- ResultT.fromResult(forSingleScienceObservingModes(observingModeName, explicitAcquisition, explicitScience, newRequirement, c, which))
-              a  = m.toList.map { case (oid, (acq, _)) => oid -> acq }
-              s  = m.toList.map { case (oid, (_, sci)) => oid -> sci }
-              _ <- ResultT.liftF(insert(a, ExposureTimeModeRole.Acquisition))
-              _ <- ResultT.liftF(insert(s, ExposureTimeModeRole.Science))
-            yield ()).value
+      )(using Transaction[F]): F[Result[Map[Observation.Id, (ExposureTimeMode, NonEmptyList[(A, ExposureTimeMode)])]]] =
+
+        val science: F[Result[Map[Observation.Id, NonEmptyList[(A, ExposureTimeMode)]]]] =
+          selectRequirement(which).map: curRequirements =>
+            val sci =
+              which
+                .fproduct: oid =>
+                  explicitScience.traverse: tup =>
+                    tup.traverse: otm =>
+                      otm orElse newRequirement orElse curRequirements.get(oid)
+                .collect:
+                  case (oid, Some(lst)) => oid -> lst
+                .toMap
+
+            val missing = which.toSet -- sci.keySet
+
+            def error: Problem =
+              val msg  = s"${observingModeName} requires a science exposure time mode."
+              OdbError.InvalidArgument(msg.some).asProblem
+
+            Result.fromOption(Option.when(missing.isEmpty)(sci), error)
+
+        ResultT(science)
+          .map: m =>
+           // Add acquisition ETM.  When there are multiple science ETMs we
+           // just pick the first for the `at`.
```

Comment:
Yeah it's a bit dodgy.  I'm guessing this will need to change at some point.  I'm not sure what the imaging acquisition exposure time mode situation is yet.